nucleus/dataset_item.py

```diff
@@ -195,13 +195,15 @@ def check_all_paths_remote(dataset_items: Sequence[DatasetItem]):
 def check_for_duplicate_reference_ids(dataset_items: Sequence[DatasetItem]):
     ref_ids = []
     for dataset_item in dataset_items:
-        if dataset_item.reference_id is not None:
-            ref_ids.append(dataset_item.reference_id)
+        if dataset_item.reference_id is None:
+            raise ValueError("Reference ID cannot be None")
```

Comment:
```suggestion
            raise ValueError(f"Reference ID cannot be None! Encountered DatasetItem with no reference ID:\n{dataset_item}")
```
maybe something like this to make it easier for the user to debug? assuming the stack trace won't show it